plugins/aggregate/aggregate.cpp

```diff
@@ -0,0 +1,469 @@
+                    .ValueOrDie();
+              auto append_result
+                = batch_builder->GetField(builder_column)
+                    ->AppendScalar(
+                      *min_max.scalar_as<arrow::StructScalar>().value[0]);
+              VAST_ASSERT(append_result.ok());
+              break;
+            }
+            case action::max: {
+              auto min_max
+                = arrow::compute::MinMax(bucket.second[builder_column])
+                    .ValueOrDie();
+              auto append_result
+                = batch_builder->GetField(builder_column)
+                    ->AppendScalar(
+                      *min_max.scalar_as<arrow::StructScalar>().value[1]);
+              VAST_ASSERT(append_result.ok());
+              break;
+            }
+            case action::mean: {
+              auto mean = arrow::compute::Mean(bucket.second[builder_column])
+                            .ValueOrDie();
+              auto cast_result
+                = mean.scalar()->CastTo(bucket.second[builder_column]->type());
+              VAST_ASSERT(cast_result.ok());
+              auto append_result
+                = batch_builder->GetField(builder_column)
+                    ->AppendScalar(*cast_result.MoveValueUnsafe());
+              VAST_ASSERT(append_result.ok());
+              break;
+            }
+            case action::any: {
+              auto any = arrow::compute::Any(bucket.second[builder_column])
+                           .ValueOrDie();
+              auto append_result = batch_builder->GetField(builder_column)
+                                     ->AppendScalar(*any.scalar());
+              VAST_ASSERT(append_result.ok());
+              break;
+            }
+            case action::all: {
+              auto any = arrow::compute::All(bucket.second[builder_column])
+                           .ValueOrDie();
+              auto append_result = batch_builder->GetField(builder_column)
+                                     ->AppendScalar(*any.scalar());
+              VAST_ASSERT(append_result.ok());
+              break;
+            }
+          }
+          ++builder_column;
+        }
+      }
+      // Create aggregated record batch, and emplace it into the result
+      // alongside the modified type.
+      auto adjusted_batch = std::shared_ptr<arrow::RecordBatch>{};
+      auto flush_result = batch_builder->Flush(&adjusted_batch);
+      VAST_ASSERT(flush_result.ok());
+      result.emplace_back(adjusted_layout, adjusted_batch);
+    }
+    return result;
+  }
+
+private:
+  /// The underlying configuration of the transformation.
+  configuration config_ = {};
+
+  /// Cache for mapping of layout to resolved columns.
+  std::unordered_map<type, resolved_columns> resolved_columns_cache_ = {};
+
+  /// Options for the `round_to_multiple` Arrow Compute function.
+  arrow::compute::RoundTemporalOptions round_temporal_options_{
+    1, arrow::compute::CalendarUnit::SECOND};
```

Comment:
We should document clearly that we always use seconds as the base unit, because using nanoseconds here is broken in at least the Arrow 7.0.0 available via Homebrew on macOS.

Reply:
milliseconds also works, and I think that is fine enough for rounding that we don't need to explain it.